src/components/SearchBar/index.js

```diff
@@ -0,0 +1,21 @@
+import React, {useState} from 'react'
+
+function SearchBar({setUsername}) {
+    const [ nameInput, setNameInput ] = useState("");
+    
+    const handleSubmit = e => {
+        e.preventDefault()
+        setUsername(nameInput)
+        e.target.elements.namedItem('username').value = ''
+    }
+
+    return (
+        <form id= "form" onSubmit={handleSubmit}>
+            {/* <label  htmlFor='username'>Username</label> */}
+            <input type="text" id='username' placeholder='Username' onChange={e => setNameInput(e.target.value)} />
```

Comment:
use value={nameInput} to to have better control over what the user is able to see in the search bar, and just clear the search with setNameInput(''). This way you are able to control the state with the onChange function and change what is possible for the user to search and see.